File: TwitterElectron/TwitterElectron/App/OptionsForm.js
var Electron = require("electron");

Bridge.assembly("TwitterElectron", function ($asm, globals) {
    "use strict";

    Bridge.define("TwitterElectron.RendererProcess.OptionsForm", {
        main: function Main () {
            var okButton = document.getElementById("okButton");
            var cancelButton = document.getElementById("cancelButton");

            Electron.ipcRenderer.on("cmd-options-restore", function (ev, cred) {
                document.getElementById("apiKeyInput").value = cred != null ? cred.ApiKey : null;
                document.getElementById("apiSecretInput").value = cred != null ? cred.ApiSecret : null;
                document.getElementById("accessTokenInput").value = cred != null ? cred.AccessToken : null;
                document.getElementById("accessTokenSecretInput").value = cred != null ? cred.AccessTokenSecret : null;
            });

            okButton.addEventListener("click", function () {
                var cred = { ApiKey: document.getElementById("apiKeyInput").value, ApiSecret: document.getElementById("apiSecretInput").value, AccessToken: document.getElementById("accessTokenInput").value, AccessTokenSecret: document.getElementById("accessTokenSecretInput").value };

                Electron.ipcRenderer.send("cmd-options-updated", cred);
                Electron.remote.getCurrentWindow().close();
            });

            cancelButton.addEventListener("click", function () {
                Electron.remote.getCurrentWindow().close();
            });
        },
        statics: {
            fields: {
                Electron: null
            }
        }
    });
});

File: TwitterElectron/TwitterElectron/RendererProcess/OptionsForm.cs
using System;
using Bridge;
using static Retyped.dom;
using static Retyped.electron;
using static Retyped.node;

namespace TwitterElectron.RendererProcess
{
    public class OptionsForm
    {
        [Init(InitPosition.Top)]
        public static void InitGlobals()
        {
            var Electron = (Electron.AllElectron)require.Self("electron");
        }

        [Template("Electron")]
        public static Electron.AllElectron Electron;

        public static void Main()
        {
            var okButton = document.getElementById("okButton");
            var cancelButton = document.getElementById("cancelButton");

            Electron.ipcRenderer.on(Constants.IPC.RestoreOptions, new Action<Event, TwitterCredentials>((ev, cred) =>
            {
                document.getElementById("apiKeyInput").As<HTMLInputElement>().value = cred?.ApiKey;
                document.getElementById("apiSecretInput").As<HTMLInputElement>().value = cred?.ApiSecret;
                document.getElementById("accessTokenInput").As<HTMLInputElement>().value = cred?.AccessToken;
                document.getElementById("accessTokenSecretInput").As<HTMLInputElement>().value = cred?.AccessTokenSecret;
            }));

            okButton.addEventListener("click", () => 
            {
                var cred = new TwitterCredentials
                {
                    ApiKey = document.getElementById("apiKeyInput").As<HTMLInputElement>().value,
                    ApiSecret = document.getElementById("apiSecretInput").As<HTMLInputElement>().value,
                    AccessToken = document.getElementById("accessTokenInput").As<HTMLInputElement>().value,
                    AccessTokenSecret = document.getElementById("accessTokenSecretInput").As<HTMLInputElement>().value
                };

                Electron.ipcRenderer.send(Constants.IPC.OptionsUpdated, cred);
                Electron.remote.getCurrentWindow().close();
            });

            cancelButton.addEventListener("click", () =>
            {
                Electron.remote.getCurrentWindow().close();
            });
        }
    }
}